server/blob/blob-storage-strategy/src/main/java/org/apache/james/server/blob/deduplication/MinIOGenerationAwareBlobId.java

```diff
@@ -64,23 +68,29 @@ public BlobId parse(String id) {
                 int family = Integer.parseInt(id.substring(0, separatorIndex1));
                 int generation = Integer.parseInt(id.substring(separatorIndex1 + 1, separatorIndex2));
                 String blobIdPart = id.substring(separatorIndex2 + 1);
-                BlobId wrapped = delegate.parse(blobIdPart.charAt(0) + "/" +
-                    blobIdPart.charAt(1) + "/" +
-                    blobIdPart.charAt(2) + "/" +
-                    blobIdPart.charAt(3) + "/" +
-                    blobIdPart.substring(4));
```

Comment:
Sorry I should have had put this in `of`. Not in parse.

I will update base PR